docs/configuration.html

```diff
@@ -308,6 +308,157 @@ <h3 class="anchor-heading"><a id="tieredstorageconfigs" class="anchor-link"></a>
+    You can then package the implementation as a JAR file, add the JAR to your classpath, and reference the provider's class in your configuration.
+  </p>
+  
+  <p><b>Example custom provider configuration</b></p>
+  <pre><code>config.providers=customProvider
+config.providers.customProvider.class=com.example.customProvider
+config.providers.customProvider.param.param1=value1
+config.providers.customProvider.param.param2=value2</code></pre>
+  
+  <h4 class="anchor-heading">
+    <a id="directory_config_provider" class="anchor-link"></a>
+    <a href="#directory_config_provider">DirectoryConfigProvider</a>
+  </h4>
+  
+  <p>
+    The <code>DirectoryConfigProvider</code> retrieves configuration data from files stored in a specified directory.
+  </p>
+  <p>
+    Each file represents a key, and its content is the value. You specify the path to the directory where the configuration files are stored using the <code>dir</code> parameter. This provider is useful for loading multiple configuration files and for organizing configuration data into separate files.
+  </p>
+  
+  <p><b>Example <code>DirectoryConfigProvider</code> configuration</b></p>
+  <pre><code>config.providers=dirProvider
+config.providers.dirProvider.class=org.apache.kafka.common.config.provider.DirectoryConfigProvider
+config.providers.dirProvider.param.dir=/path/to/config/directory</code></pre>
+  
+  <h4 class="anchor-heading">
+    <a id="env_var_config_provider" class="anchor-link"></a>
+    <a href="#env_var_config_provider">EnvVarConfigProvider</a>
+  </h4>
+  
+  <p>The <code>EnvVarConfigProvider</code> retrieves configuration data from environment variables.</p>
+  <p>No specific parameters are required, as it reads directly from the specified environment variables.</p>
+  <p>
+    This provider is useful for configuring applications running in containers, for example, to load certificates or JAAS configuration from environment variables mapped from secrets.
+  </p>
+  
+  <p><b>Example <code>EnvVarConfigProvider</code> configuration</b></p>
+  <pre><code>config.providers=envVarProvider
+config.providers.envVarProvider.class=org.apache.kafka.common.config.provider.EnvVarConfigProvider</code></pre>
+  
+  <h4 class="anchor-heading">
+    <a id="file_config_provider" class="anchor-link"></a>
+    <a href="#file_config_provider">FileConfigProvider</a>
+  </h4>
+  
+  <p>The <code>FileConfigProvider</code> retrieves configuration data from a single properties file.</p>
+  <p>This provider is useful for loading configuration data from mounted files.</p>
+  
+  <p><b>Example <code>FileConfigProvider</code> configuration</b></p>
+  <pre><code>config.providers=fileProvider
+config.providers.fileProvider.class=org.apache.kafka.common.config.provider.FileConfigProvider</code></pre>
+  
+  <h4 class="anchor-heading">
+    <a id="ref_config_provider" class="anchor-link"></a>
+    <a href="#ref_config_provider">Referencing configuration providers</a>
```

Comment:
I wonder if it would read better if the syntax to use each provider were in their respective sections? WDYT?

Reply:
My thinking is that in most cases (configuring clients, brokers, Streams applications, the Connect runtime), it's the same person doing both actions and doing them at the same time. The only outlier scenario is using configuration providers in a connector configuration as the provider first needs to be defined in the runtime and used at a later time when creating a connector.